lib/mix/tasks/eunit.ex

```diff
@@ -36,25 +36,36 @@ defmodule Mix.Tasks.Eunit do
 
   """
 
+  @cover [output: "cover", tool: Mix.Tasks.Test.Cover]
+
   def run(args) do
     options = parse_options(args)
+    project = Mix.Project.config
 
     # add test directory to compile paths and add
     # compiler options for test
-    post_config = eunit_post_config(Mix.Project.config)
+    post_config = eunit_post_config(project)
     modify_project_config(post_config)
 
     # make sure mix will let us run compile
     ensure_compile
     Mix.Task.run "compile"
 
+    # start cover
+    cover =
+      if options[:cover] do
+        compile_path = Mix.Project.compile_path(project)
+        cover = Keyword.merge(@cover, project[:test_coverage] || [])
+        cover[:tool].start(compile_path, cover)
+      end
+
     # run the actual tests
-    if(options[:cover], do: cover_start())
     test_modules(post_config[:erlc_paths], options[:patterns])
     |> Enum.map(&module_name_from_path/1)
     |> Enum.drop_while(fn(m) ->
       tests_pass?(m, options[:eunit_opts] ++ post_config[:eunit_opts]) end)
-    if(options[:cover], do: cover_analyse())
+
+    cover && cover.()
```

Comment:
OK, so I think this is just a bit of funkyness with variable names (like having a couple things named `cover`).  Can we do something like

```
def run(args) do
   # .. existing code

  # line 55
  cover_callback = start_cover_tool(options[:cover], project)

  # lines 63-66 same

  # line 68
  execute_cover_callback(cover_callback)
end

# no cover tool specified
defp start_cover_tool(nil, _project), do: nil
# set up the cover tool
defp start_cover_tool(cover_tool, project) do
  compile_path = Mix.Project.compile_path(project)
  cover = Keyword.merge(@cover, project[:test_coverage] || [])
  # returns a callback
  cover[:tool].start(compile_path, cover)
end

# no cover tool was specified
defp execute_cover_callback(nil), do: :ok
# run the cover callback
defp execute_cover_callback(cb), do: cb.()
```